frontend/packages/gitops-plugin/src/components/details/GitOpsResourcesSection.tsx

```diff
@@ -0,0 +1,158 @@
+import * as React from 'react';
+import {
+  Stack,
+  StackItem,
+  Split,
+  SplitItem,
+  Card,
+  CardBody,
+  Tooltip,
+} from '@patternfly/react-core';
+import { HeartBrokenIcon } from '@patternfly/react-icons';
+import { useTranslation } from 'react-i18next';
+import { ResourceIcon } from '@console/internal/components/utils';
+import { GitOpsEnvironmentService, GitOpsHealthResources } from '../utils/gitops-types';
+import './GitOpsResourcesSection.scss';
+
+interface GitOpsResourcesSectionProps {
+  services: GitOpsEnvironmentService[];
+  secrets: GitOpsHealthResources[];
+  deployments: GitOpsHealthResources[];
+  routes: GitOpsHealthResources[];
+  roleBindings: GitOpsHealthResources[];
+  clusterRoles: GitOpsHealthResources[];
+  clusterRoleBindings: GitOpsHealthResources[];
+}
+
+const getUnhealthyResources = () => (acc: string[], current: GitOpsHealthResources): string[] =>
+  current.health === 'Degraded' ||
+  current.health === 'Progressing' ||
+  current.health === 'Missing' ||
+  current.health === 'Unknown'
+    ? [...acc, current.health]
+    : acc;
+
+const getNonSyncedResources = () => (acc: string[], current: GitOpsHealthResources): string[] =>
+  current.status !== 'Synced' ? [...acc, current.status] : acc;
+
+const GitOpsResourcesSection: React.FC<GitOpsResourcesSectionProps> = ({
+  services,
+  secrets,
+  deployments,
+  routes,
+  roleBindings,
+  clusterRoles,
+  clusterRoleBindings,
+}) => {
+  const { t } = useTranslation();
+  const degradedServices: string[] = services ? services.reduce(getUnhealthyResources(), []) : [];
+  const degradedDeployments: string[] = deployments
+    ? deployments.reduce(getUnhealthyResources(), [])
+    : [];
+  const degradedSecrets: string[] = secrets ? secrets.reduce(getUnhealthyResources(), []) : [];
+  const degradedRoutes: string[] = routes ? routes.reduce(getNonSyncedResources(), []) : [];
+  const degradedRoleBindings: string[] = roleBindings
+    ? roleBindings.reduce(getNonSyncedResources(), [])
+    : [];
+  const degradedClusterRoles: string[] = clusterRoles
+    ? clusterRoles.reduce(getNonSyncedResources(), [])
+    : [];
+  const degradedClusterRoleBindings: string[] = clusterRoleBindings
+    ? clusterRoleBindings.reduce(getNonSyncedResources(), [])
+    : [];
+
+  const renderResource = (
```

Comment:
Might want to make this a component by itself. Not sure there is a lot of gain making it a function. Easier to maintain if it's a stand-alone component that just takes the two props.

Reply:
Added to issue [GITOPS-1279](https://issues.redhat.com/browse/GITOPS-1279)